includes/Services/AI_Service.php

```diff
@@ -0,0 +1,231 @@
+<?php
+/**
+ * AI Service implementation.
+ *
+ * Provides a centralized service layer for AI operations.
+ *
+ * @package WordPress\AI\Services
+ */
+
+declare( strict_types=1 );
+
+namespace WordPress\AI\Services;
+
+use WordPress\AI_Client\AI_Client;
+use WordPress\AI_Client\Builders\Prompt_Builder_With_WP_Error;
+use WordPress\AiClient\Providers\Models\DTO\ModelConfig;
+
+use function WordPress\AI\get_preferred_models;
+
+/**
+ * AI Service class.
+ *
+ * Manages AI provider configuration and provides a consistent interface
+ * for experimental features to communicate with AI providers.
+ *
+ * @since 0.1.0
+ */
+class AI_Service {
+
+	/**
+	 * Singleton instance.
+	 *
+	 * @since 0.1.0
+	 *
+	 * @var \WordPress\AI\Services\AI_Service|null
+	 */
+	private static ?self $instance = null;
+
+	/**
+	 * Whether the service has been initialized.
+	 *
+	 * @since 0.1.0
+	 *
+	 * @var bool
+	 */
+	private bool $initialized = false;
+
+	/**
+	 * Option key mapping from WordPress snake_case to SDK camelCase.
+	 *
+	 * @since 0.1.0
+	 *
+	 * @var array<string, string>
+	 */
+	private static array $option_key_map = array(
+		'system_instruction' => 'systemInstruction',
+		'candidate_count'    => 'candidateCount',
+		'max_tokens'         => 'maxTokens',
+		'temperature'        => 'temperature',
+		'top_p'              => 'topP',
+		'top_k'              => 'topK',
+		'stop_sequences'     => 'stopSequences',
+		'presence_penalty'   => 'presencePenalty',
+		'frequency_penalty'  => 'frequencyPenalty',
+		'logprobs'           => 'logprobs',
+		'top_logprobs'       => 'topLogprobs',
+	);
+
+	/**
+	 * Gets the singleton instance.
+	 *
+	 * @since 0.1.0
+	 *
+	 * @return \WordPress\AI\Services\AI_Service The singleton instance.
+	 */
+	public static function get_instance(): self {
+		if ( null === self::$instance ) {
+			self::$instance = new self();
+		}
+		return self::$instance;
+	}
+
+	/**
+	 * Private constructor to enforce singleton pattern.
+	 *
+	 * @since 0.1.0
+	 */
+	private function __construct() {}
+
+	/**
+	 * Initializes the AI service.
+	 *
+	 * This method should be called after AI_Client::init() on the WordPress 'init' hook.
+	 *
+	 * @since 0.1.0
+	 */
+	public function init(): void {
+		if ( $this->initialized ) {
+			return;
+		}
+
+		$this->initialized = true;
+
+		/**
+		 * Fires when the AI service is initialized.
+		 *
+		 * @since 0.1.0
+		 *
+		 * @param \WordPress\AI\Services\AI_Service $service The AI service instance.
+		 */
+		do_action( 'ai_service_initialized', $this );
+	}
+
+	/**
+	 * Checks if an AI provider is available and configured.
+	 *
+	 * @since 0.1.0
+	 *
+	 * @return bool True if a provider is available, false otherwise.
+	 */
+	public function is_available(): bool {
```

Comment:
Do we need this method? I understand the point is to return whether _any_ AI is generally available - which _maybe_ makes sense, but I am wary because it's not going to be truly accurate.

The WordPress AI Client already offers a much more precise way of determining that, via the `is_supported*` methods of `Prompt_Builder`.

The code example included with the `get_ai_service()` documentation confirms that including this method could be misleading: It's NOT how one should check whether AI is available.